compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -94,7 +94,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
     lazy val accessors =
       if (isDerivedValueClass(clazz)) clazz.paramAccessors.take(1) // Tail parameters can only be `erased`
       else clazz.caseAccessors
-    val isEnumCase = clazz.derivesFrom(defn.EnumClass) && clazz != defn.EnumClass
+    val isEnumCase = clazz.classParents.exists(_.typeSymbol.isScalaEnum)
```

Comment:
That does not work if the parent is an alias of an enum class. It's better to use `_.classSymbol` to avoid that trap.

Reply:
Thank you, I've updated this